iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp

```diff
@@ -203,6 +317,18 @@ struct ConvertHLOToLinalgOnTensorsPass
 
     ConversionTarget target(getContext());
     target.addIllegalDialect<mhlo::MhloDialect>();
+
+    // TODO(hanchung): Do it in a cleaner way.
+    // We don't see complex types in codegen. This assumes that we run
```

Comment:
One idea is to change the pattern above to look for users that are `mhlo.abs`and replace it a `linalg.generic` that takes the real and imaginary parts and compute the absolute.

Reply:
Good point, I think we are already doing it in this way. There are patterns to rewrite `mhlo.abs` (which takes complex typese) to `mhlo.real + mhlo.imag` to compute the magnitude in HLOToHLOPreprocessing pass.